### ISSUE TITLE
Anonymous Functions from IE are set as undefined

### ISSUE THREAD

**Issue by auchenberg** [stacktracejs/error-stack-parser#19]
I'm currently trying to normalize stack frame function names across javascript engines, by building on top of this library.

Anonymous functions from IE stack frames are filtered out, and the `functionName` property on the `StackFrame` object is set to undefined in https://github.com/stacktracejs/error-stack-parser/blob/master/error-stack-parser.js#L65

Knowing that a given function is anonymous is valuable information and shouldn't be filtered out.

Current output:

```
{ 
    fileName: 'http://localhost:8080/test.html',
    lineNumber: 20,
    columnNumber: 9,
    source: '          at Anonymous function (http://localhost:8080/test.html:20:9)'
},
```

Expect output:

```
{ 
    functionName: 'Anonymous function',
    fileName: 'http://localhost:8080/test.html',
    lineNumber: 20,
    columnNumber: 9,
    source: '          at Anonymous function (http://localhost:8080/test.html:20:9)'
},
```

**Comment by eriwen** [comment]
Currently we intentionally omit anonymous functions because different browsers use different terms for them (the intent is some amount of normalization). That said, perhaps there's value is providing a `const`ant way this library represents an anonymous function other than `undefined`? 

If on the other hand, the bug is that in other browsers we have a function name of `"Anonymous function"` and IE is an outlier then normalizing this way may be the right thing to do.

My question to you @auchenberg is: Does representing "Anonymous function" this way help your error-stack-normalizer? When is this preferable to using a false-y value?

**Comment by auchenberg** [comment]
Yes, a representation of a "anonymous function" does indeed help my error-stack-normalizer. 

With the current logic `undefined` represents both anonymous and unparsable function names, which makes normalization of anonymous-function across engines somewhat hard :wink: 

I don't think it should be the responsibly of this library to normalize value of anonymous functions, but only focus on what the name implies: parsing of the various stack frame formats.

**Comment by eriwen** [comment]
Okay, that makes sense. I would like to reduce cases where function names are un-parseable. Do you have any examples. 

Seems like we should make sure all browsers represent "Anonymous" the same way. Would you suggest "Anonymous function" or something else?

**Comment by eriwen** [comment]
Suppose we didn't do any pre-processing as in #23? Maybe that would be more kind to your normalization efforts @auchenberg

**Comment by auchenberg** [comment]
@eriwen That looks good. My normalization efforts are bit different, as it involves rewriting function names, etc, so generate a normalized footprint across the engines in addition to normalize anonymous function names.
